### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by LeeLorenzSr**
This pull request introduces several improvements and refactorings across the codebase, focusing on code clarity, test coverage, and minor behavioral changes. The most significant updates include expanded and enhanced unit tests for the `IniFile` class, code style and consistency improvements in both implementation and helper methods, and updated dependencies for the test project.

**Unit test improvements and expanded coverage:**

* Added multiple new unit tests in `IniFileTests.cs` to cover edge cases such as comment handling, duplicate keys, error conditions, options honoring, and utility methods. Tests now also verify proper handling of comments, warnings, and special options. [[1]](diffhunk://#diff-08dd8982f92f82fd30df2000d14e8a849f03718af691486ea129556b058f99f8L19-R51) [[2]](diffhunk://#diff-08dd8982f92f82fd30df2000d14e8a849f03718af691486ea129556b058f99f8L49-R153)
* Refactored variable declarations in tests to use `var` for consistency and modern C# style. [[1]](diffhunk://#diff-08dd8982f92f82fd30df2000d14e8a849f03718af691486ea129556b058f99f8L10-R12) [[2]](diffhunk://#diff-08dd8982f92f82fd30df2000d14e8a849f03718af691486ea129556b058f99f8L49-R153) [[3]](diffhunk://#diff-08dd8982f92f82fd30df2000d14e8a849f03718af691486ea129556b058f99f8L84-R168) [[4]](diffhunk://#diff-08dd8982f92f82fd30df2000d14e8a849f03718af691486ea129556b058f99f8L95-L99)

**Code style, clarity, and helper refactoring:**

* Refactored various helper methods in `IniHelpers.cs` to use expression-bodied members, improved naming, and simplified logic. Also updated documentation to accurately reflect method behavior (e.g., line splitting no longer trims lines). [[1]](diffhunk://#diff-0db129e4bf98c5c277ae2f4f9de8d176a3ac5fc1934b641fa2d833a0e8b39f6fL21-R39) [[2]](diffhunk://#diff-0db129e4bf98c5c277ae2f4f9de8d176a3ac5fc1934b641fa2d833a0e8b39f6fL54-R49) [[3]](diffhunk://#diff-0db129e4bf98c5c277ae2f4f9de8d176a3ac5fc1934b641fa2d833a0e8b39f6fL65-R80)
* Updated string handling and comparison logic to use `string.Empty`, `ToLowerInvariant()`, and modern C# idioms for improved clarity and correctness. [[1]](diffhunk://#diff-9415b01eed3091accec60800cb1dc9357d9fee893bf9db940c5d0c5ec6ce831eL48-R41) [[2]](diffhunk://#diff-9415b01eed3091accec60800cb1dc9357d9fee893bf9db940c5d0c5ec6ce831eL58-R51) [[3]](diffhunk://#diff-0db129e4bf98c5c277ae2f4f9de8d176a3ac5fc1934b641fa2d833a0e8b39f6fL127-R117) [[4]](diffhunk://#diff-0db129e4bf98c5c277ae2f4f9de8d176a3ac5fc1934b641fa2d833a0e8b39f6fL149-R139)
* Streamlined and clarified several methods in `IniFile.cs`, including `UnescapeString`, `Clear`, `CompareTo`, and various `Get` overloads, to use more concise and idiomatic C# constructs. [[1]](diffhunk://#diff-9415b01eed3091accec60800cb1dc9357d9fee893bf9db940c5d0c5ec6ce831eL24-R24) [[2]](diffhunk://#diff-9415b01eed3091accec60800cb1dc9357d9fee893bf9db940c5d0c5ec6ce831eL88-R86) [[3]](diffhunk://#diff-9415b01eed3091accec60800cb1dc9357d9fee893bf9db940c5d0c5ec6ce831eL108-R109) [[4]](diffhunk://#diff-9415b01eed3091accec60800cb1dc9357d9fee893bf9db940c5d0c5ec6ce831eL128-R123) [[5]](diffhunk://#diff-9415b01eed3091accec60800cb1dc9357d9fee893bf9db940c5d0c5ec6ce831eL148-R141) [[6]](diffhunk://#diff-9415b01eed3091accec60800cb1dc9357d9fee893bf9db940c5d0c5ec6ce831eL194-R187)

**Dependency updates:**

* Updated NuGet package versions in the test project to ensure compatibility with newer tooling and test runners.

**Documentation and Azure Copilot instructions:**

* Added Azure-specific Copilot instructions to `.github/copilot-instructions.md` to guide tool usage and best practices when handling Azure-related requests.

### CODE REVIEW

File: IniGetter/IniFile.cs [hunk 1]
```diff
@@ -105,14 +92,21 @@ public void Clear()
...
             }
-            else if (obj.GetType() == typeof(string))
+
+            if (obj is string value)
             {
-                return this.ToString().CompareTo(obj as string);
+                return ToString().CompareTo(value);
             }
+
             return -1;
```
[the remark this answers is not in the document]
Comment (LeeLorenzSr) [reply IniGetter/IniFile.cs:110]: Throwing hee would alter the original contract. Leaving as is.
